pkg/gui/context/filtered_list.go

```diff
@@ -41,6 +41,10 @@ func (self *FilteredList[T]) ClearFilter() {
 	self.SetFilter("")
 }
 
+func (self *FilteredList[T]) ReApplyFilter() {
```

Comment:
Exposing `applyFilter` publicly would be fine with me (I'd rename it to ReApplyFilter then), but on the other hand the one-line forwarder doesn't bother me much. So no strong opinion here.

I wouldn't rename it to `InvalidateModel` though, this would sound misleading to me. Maybe `OnModelInvalidated`, but I'm not sure why that's better.

The other question is more interesting: should we move some of the logic specific to contexts (e.g. selecting lines etc) from SearchHelper to FilteredListViewModel. The same could be asked for the existing code in `OnPromptContentChanged`. I don't really have much of an opinion here, happy to follow whatever Jesse tells me to do. Personally I'd leave everything the way it is, as that's the least amount of work. 😄

Reply:
Happy to leave things where they are for now, but happy to accept a separate PR that does the refactor if anybody's up to it